### PR TITLE
Update webhook signing to show JWT signing

### PR DESCRIPTION
**Pull request by bjornj12**
Updating the webhook signing docs to explain how to generate a 
secret and verify JWT auth token instead of the custom avo-signature we initially decided on.

### CODE REVIEW

**Comment by hugihlynsson** [comment]
#### 📝 Changed routes:


| [**…g-plan/webhook-signing**](https://docs-ivcnlbtbo.avo-dev.app/workspace/tracking-plan/webhook-signing) |
|:-:|
|<a href="https://docs-ivcnlbtbo.avo-dev.app/workspace/tracking-plan/webhook-signing">Screenshot of /workspace/tracking-plan/webhook-signing</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/Gb4p3DZdh3cie4Td726_9VhSSHO7l5J2NYXESprABAIEJFqkXWNj9FiBVtDXv-f8kPGLUCqX_xR9vCrkY17H7w.png">(view full size)</a> |







Commit 4cc036451b7b95014ea6f18d8d83284241c5c47c (https://docs-ivcnlbtbo.avo-dev.app).

File: pages/workspace/tracking-plan/webhook-signing.mdx [hunk 1]
```diff
@@ -11,84 +11,56 @@ import Link from '../../../components/Link';
 
 ### Verify the that only Avo is sending you events!
 
-Avo will sign each webhook call it sends to your endpoint by including a signature in each request `Avo-Signature` header. 
+Avo will sign each webhook call it sends to your endpoint by including a [Jwt](https://Jwt.io) token in each request `authorization` header.
```
Comment (logason) [inline pages/workspace/tracking-plan/webhook-signing.mdx:14]:
```suggestion
Avo will sign each webhook call it sends to your endpoint by including a [JWT](https://Jwt.io) token in each request `authorization` header. 
```

File: pages/workspace/tracking-plan/webhook-signing.mdx [hunk 2]
```diff
@@ -11,84 +11,56 @@ import Link from '../../../components/Link';
 
 ### Verify the that only Avo is sending you events!
 
-Avo will sign each webhook call it sends to your endpoint by including a signature in each request `Avo-Signature` header. 
+Avo will sign each webhook call it sends to your endpoint by including a [Jwt](https://Jwt.io) token in each request `authorization` header. 
 This allows you to verify that the events were sent by Avo and not by a third party. 
 Before you can verify signatures, you need to retrieve your publishing webhook's secret from your Webhook publishing page in Avo.
 Avo generates a unique secret key for each publishing webhook endpoint. If you use multiple endpoints, you must obtain a secret for each one you want to verify the signature on.
```
Comment (logason) [inline pages/workspace/tracking-plan/webhook-signing.mdx:17]:
```suggestion
Avo generates a unique secret key for each publishing webhook endpoint. If you use multiple webhooks, you must obtain a secret for each one you want to verify the signature for.
```

File: pages/workspace/tracking-plan/webhook-signing.mdx [hunk 3]
```diff
@@ -11,84 +11,56 @@ import Link from '../../../components/Link';
...
+    alt="The image shows a webhook integration and a button at the top labeled Webhook Secret which if clicked reveals the secret"
 />
 </p>
 
 
 ### Verifying signatures
 
-The `Avo-signature` header included in each request contains a timestamp and a signature. 
-The timestamp(Unix timestamp) is prefixed by `ts=` and the signature is prefixed by `v1`.
+The `authorization` header included in each request contains a Jwt token that can be used to validate the request using the secret given in the webhook configuration dashboard.
```
Comment (logason) [inline pages/workspace/tracking-plan/webhook-signing.mdx:29]:
```suggestion
The `authorization` header included in each request contains a JWT token that can be used to validate the request using the secret given in the webhook configuration dashboard.
```

File: pages/workspace/tracking-plan/webhook-signing.mdx [hunk 4]
```diff
@@ -11,84 +11,56 @@ import Link from '../../../components/Link';
...
 
-```json
-Avo-Signature:ts=1651677541,v1=30703093db3c532c6f0ab1f1f9eb80b134f5bf6e2ff7064c403aab7582df0667
-```
-
-<br />
 
 Avo generates signatures using a hash-based message authentication code (HMAC) with SHA-256.
```
Comment (logason) [inline pages/workspace/tracking-plan/webhook-signing.mdx:32]: This still true?

File: pages/workspace/tracking-plan/webhook-signing.mdx [hunk 5]
```diff
@@ -11,84 +11,56 @@ import Link from '../../../components/Link';
...
- ```pseudocode
-  var signed_payload = 
-  requestTimestamp // Timestamp from Avo-Signature header
-  ++ "." 
-  ++ Json.stringify(request.body);
- ```
+#### 2. Validate the authorization header
 
-#### 3. Determine the expected signature
+Using a [Jwt library](https://Jwt.io/libraries), verify that the token is valid.
```
Comment (logason) [inline pages/workspace/tracking-plan/webhook-signing.mdx:48]:
```suggestion
Using a [JWT library](https://Jwt.io/libraries), verify that the token is valid.
```

File: pages/workspace/tracking-plan/webhook-signing.mdx [hunk 6]
```diff
@@ -11,84 +11,56 @@ import Link from '../../../components/Link';
...
+    var token = request.headers.get('authorization');
+    token = token.split(" ")[1]; //Remove "Bearer " from the token
+    Jwt.verify(token, _secret); // Some libraries require try catch around this.
+ ```
 
-#### 4. Compare the signatures
-Now that you have hashed the secret with the `signed_payload`, you can now compare it with the signature from the `Avo-Signature` header.
+#### 3. Extra security (optional)
 
+Each Jwt token has in it's payload an "iat"(Issued at time) which is a timestamp you can use for extra security to match with the current timestamp. That way you can make sure the timestamp in the request is not very old.
```
Comment (logason) [inline pages/workspace/tracking-plan/webhook-signing.mdx:60]:
```suggestion
Each JWT token has in it's payload an "iat"(Issued at time) which is a timestamp you can use for extra security to match with the current timestamp. That way you can make sure the timestamp in the request is not very old.
```

Comment (logason) [inline pages/workspace/tracking-plan/webhook-signing.mdx:60]: > That way you can make sure the timestamp in the request is not very old

Can we add some additional details here on why that is a good thing to do?

File: pages/workspace/tracking-plan/webhook-signing.mdx (hunk 4 above)
Comment (bjornj12) [reply pages/workspace/tracking-plan/webhook-signing.mdx:32]: yes

File: pages/workspace/tracking-plan/webhook-signing.mdx [hunk 7]
```diff
@@ -9,86 +9,64 @@ import Link from '../../../components/Link';
 
 ## Check the webhook signatures
 
-### Verify the that only Avo is sending you events!
+### Verify the that only Avo is sending you events
```
Comment (alex-tpom6oh) [inline pages/workspace/tracking-plan/webhook-signing.mdx:12]:
```suggestion
### Verify that only Avo is sending you events
```

File: pages/workspace/tracking-plan/webhook-signing.mdx [hunk 8]
```diff
@@ -9,86 +9,64 @@ import Link from '../../../components/Link';
...
 
-Avo will sign each webhook call it sends to your endpoint by including a signature in each request `Avo-Signature` header. 
+Avo will sign each webhook call it sends to your endpoint by including a [JWT](https://Jwt.io) token in each request `authorization` header. 
 This allows you to verify that the events were sent by Avo and not by a third party. 
 Before you can verify signatures, you need to retrieve your publishing webhook's secret from your Webhook publishing page in Avo.
-Avo generates a unique secret key for each publishing webhook endpoint. If you use multiple endpoints, you must obtain a secret for each one you want to verify the signature on.
+Avo generates a unique secret key for each publishing webhook endpoint. If you use multiple webhooks, you must obtain a secret for each one you want to verify the signature for.
```
Comment (alex-tpom6oh) [inline pages/workspace/tracking-plan/webhook-signing.mdx:17]:
```suggestion
Avo generates a unique secret key for each webhook publishing integration. You should obtain a distinct key for every webhook integration you use.
```

File: pages/workspace/tracking-plan/webhook-signing.mdx [hunk 9]
```diff
@@ -9,86 +9,64 @@ import Link from '../../../components/Link';
...
+    alt="The image shows a webhook integration and a button at the top labeled Webhook Secret which if clicked reveals the secret"
 />
 </p>
 
 
 ### Verifying signatures
 
-The `Avo-signature` header included in each request contains a timestamp and a signature. 
-The timestamp(Unix timestamp) is prefixed by `ts=` and the signature is prefixed by `v1`.
+The `authorization` header included in each request contains a JWT token that can be used to validate the request using the secret given in the webhook configuration dashboard.
```
Comment (alex-tpom6oh) [inline pages/workspace/tracking-plan/webhook-signing.mdx:29]:
```suggestion
The `authorization` header included in each request contains a JWT token that can be used to validate the request using the secret key given in the webhook configuration dashboard.
```

File: pages/workspace/tracking-plan/webhook-signing.mdx [hunk 10]
```diff
@@ -9,86 +9,64 @@ import Link from '../../../components/Link';
...
-    v1:30703093db3c532c6f0ab1f1f9eb80b134f5bf6e2ff7064c403aab7582df0667
+  "alg": "HS256",
+  "typ": "Jwt"
 }
 ```
-
 <br />
 
-#### 2. Prepare the signed payload
+### Verify signature header
```
Comment (alex-tpom6oh) [inline pages/workspace/tracking-plan/webhook-signing.mdx:41]: Maybe authorization header?

File: pages/workspace/tracking-plan/webhook-signing.mdx [hunk 11]
```diff
@@ -9,86 +9,64 @@ import Link from '../../../components/Link';
...
 
-The `signed_payload` string is created by concatenating
- - The timestamp (string)
- - The character `.`
- - The JSON payload (request.body)
+#### 1. Store your secret safely.
```
Comment (alex-tpom6oh) [inline pages/workspace/tracking-plan/webhook-signing.mdx:43]:
```suggestion
#### 1. Store your secret key safely.
```

File: pages/workspace/tracking-plan/webhook-signing.mdx [hunk 12]
```diff
@@ -9,86 +9,64 @@ import Link from '../../../components/Link';
...
+The secret you received in our webhook configuration, should be stored in a secret manager or an env variable on your server.
```
Comment (alex-tpom6oh) [inline pages/workspace/tracking-plan/webhook-signing.mdx:44]:
```suggestion
The secret key you received in our webhook configuration, should be stored in a secret manager or an environment variable on your server.
```

**Review by alex-tpom6oh — APPROVED** [review]
Looks good, had a couple of suggestions

**Pull request merged**